### DISCUSSION TITLE
DiscordProvider not returning all values

### DISCUSSION THREAD

**Discussion by Coow** [nextauthjs/next-auth#3425] [Help]
### Question 💬

Using the default DiscordProvider doesn't seem to be returning an object with ID, name, email and image.
I for some reason only get a object user object with, name, email and image. 

I am simply using the default provider without any additional settings

### How to reproduce ☕️

`    providers: [
        DiscordProvider({
            clientId: process.env.DISCORD_CLIENT_ID,
            clientSecret: process.env.DISCORD_CLIENT_SECRET,
        }),
    ],
`

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by zachguo** [comment]
~~yeah, `id` is missing.~~
Discord `id` can be retrieved from the `providerAccountId` field in `accounts` table.

**Reply by zoeleu** [reply]
shouldnt be, should be in the user object.

**Comment by Herdiee** [comment]
This is my workaround for getting the discord user snowflake (ID) using database sessions without using email as a scope.
```js
providers: [
        DiscordProvider({
            clientId: process.env.DISCORD_CLIENT_ID,
            clientSecret: process.env.DISCORD_CLIENT_SECRET,
            authorization: { params: { scope: 'identify' } },
            profile: (profile) => {
                let userAvatar = `https://cdn.discordapp.com/avatars/${profile.id}/${profile.avatar}.png`
                return {
                    id: profile.id,
                    snowflake: profile.id,
                    name: profile.username,
                    image: userAvatar,
                }
            },
        }),
    ],
 ```
    
I noticed that in the signIn callback, the first time the user logs in, user.id returns the users discord ID and future log ins return the default id set by `@default(cuid())` meaning you can't modify the users record to add their ID as the record isn't created. Errors are thrown when `@default` is removed, so instead I'm just using a snowflake field. 

Snowflake can be accessed in session by using the callback
```js
async session({ session, user, token }) {
            session.snowflake = user.snowflake
            return Promise.resolve(session)
 },
````
    
Not sure if this is the right way to do it but it works - might be worth expanding the docs for this provider.
